AK/Utf8View.cpp

```diff
@@ -32,24 +32,25 @@ Utf8CodePointIterator Utf8View::iterator_at_byte_offset_without_validation(size_
 
 size_t Utf8View::code_point_offset_of(size_t byte_offset) const
 {
-    VERIFY(byte_offset < byte_length());
+    VERIFY(byte_offset <= byte_length());
 
     // Fast path: each code point is represented by a single byte.
     if (length() == byte_length())
         return byte_offset;
 
     size_t code_point_offset = 0;
-    for (auto it = begin(); !it.done(); ++it) {
-        if (it.m_ptr > begin_ptr() + byte_offset)
+    auto it = begin();
+    while (!it.done()) {
```

Comment:
welp this made me realize how wrong `Utf16View::code_point_offset_of` is :sweat_smile: 

FYI we currently don't assert on the length in `Utf16Vew` because the JS spec purposefully throws an index larger than the length at it by way of `AdvanceStringIndex`:

https://tc39.es/ecma262/#sec-advancestringindex
```c++
// 4. If index + 1 ≥ length, return index + 1.
if (index + 1 >= string.length_in_code_units())
    return index + 1;
```

But maybe we should revisit that and add some protection at the caller.